pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -490,11 +490,9 @@ int lwip_sock_get_addr(struct netconn *conn, struct _sock_tl_ep *ep, u8_t local)
         ) {
         return res;
     }
-#if LWIP_IPV6 && LWIP_IPV4
-    ep->family = (addr.type == IPADDR_TYPE_V6) ? AF_INET6 : AF_INET;
-#elif LWIP_IPV6
```

Comment:
Why remove dual stack support here?

Reply:
Also why not use `IS_USED()` in this block?

Reply:
It is not removing dual stack support. In case both are enabled, IPv6 is also enabled. (The test this fixes is actually the dual stack case.)

Reply:
But if IPv6 is enabled but not IPv4, then `AF_INET` is set regardless. I know this is broken in master as well, but if neither are enabled `family` remains unset in master, which seems to me more logical.

Reply:
Is it possible to use lwIP without IP?

Reply:
> Is it possible to use lwIP without IP?

In our version it currently does not compile (probably due to the confusion mentioned [below](https://github.com/RIOT-OS/RIOT/pull/14665#issuecomment-667026558) between `IS_USED(MODULE_LWIP_IPV4)` and `IS_ACTIVE(LWIP_IPV4)`), but there shouldn't be anything preventing you from compiling lwIP without IP. Maybe you even want to, if you want to use other network layer protocols or IPv7 ;-P

Reply:
OK. I get your point. But in that case we should also check if a packet is indeed IPv4, but there is no `NETCONNTYPE_ISIPV4`. It seems to be impossible to distinguish between IPv4 and non-IP.

Reply:
True, but in that case just let's assume that if `LWIP_IPV4` is active and `!NETCONNTYPE_ISIPV6(conn)` that the family is `AF_INET`.